src/bsoncxx/types/bson_value/value.cpp

```diff
@@ -25,6 +27,162 @@ BSONCXX_INLINE_NAMESPACE_BEGIN
+    _impl->_value.value_type = BSON_TYPE_DOUBLE;
+    _impl->_value.value.v_double = v;
+}
+
+value::value(b_int32 v) : value(v.value) {}
+value::value(int32_t v) : _impl{stdx::make_unique<impl>()} {
+    _impl->_value.value_type = BSON_TYPE_INT32;
+    _impl->_value.value.v_int32 = v;
+}
+
+value::value(b_int64 v) : value(v.value) {}
+value::value(int64_t v) : _impl{stdx::make_unique<impl>()} {
+    _impl->_value.value_type = BSON_TYPE_INT64;
+    _impl->_value.value.v_int64 = v;
+}
+
+value::value(const char* v) : value(stdx::string_view{v}) {}
+value::value(std::string v) : value(stdx::string_view{v}) {}
+value::value(b_utf8 v) : value(v.value) {}
+value::value(stdx::string_view v) : _impl{stdx::make_unique<impl>()} {
+    _impl->_value.value_type = BSON_TYPE_UTF8;
+    _impl->_value.value.v_utf8.str = make_copy_for_libbson(v);
+    _impl->_value.value.v_utf8.len = (uint32_t)v.size();
+}
+
+value::value(b_null) : value(nullptr) {}
+value::value(std::nullptr_t) : _impl{stdx::make_unique<impl>()} {
+    _impl->_value.value_type = BSON_TYPE_NULL;
+}
+
+value::value(b_date v) : value(v.value) {}
+value::value(std::chrono::milliseconds v) : _impl{stdx::make_unique<impl>()} {
+    _impl->_value.value_type = BSON_TYPE_DATE_TIME;
+    _impl->_value.value.v_datetime = v.count();
+}
+
+value::value(b_oid v) : value(v.value) {}
+value::value(oid v) : _impl{stdx::make_unique<impl>()} {
+    _impl->_value.value_type = BSON_TYPE_OID;
+    std::memcpy(_impl->_value.value.v_oid.bytes, v.bytes(), v.k_oid_length);
+}
+
+value::value(b_bool v) : value(v.value) {}
+value::value(bool v) : _impl{stdx::make_unique<impl>()} {
+    _impl->_value.value_type = BSON_TYPE_BOOL;
+    _impl->_value.value.v_bool = v;
+}
+
+value::value(b_code v) : value(v.type_id, v) {}
+value::value(b_regex v) : value(v.type_id, std::string{v.regex}, std::string{v.options}) {}
+value::value(b_symbol v) : value(v.type_id, v) {}
+value::value(b_maxkey) : value(type::k_maxkey) {}
+value::value(b_minkey) : value(type::k_minkey) {}
+value::value(b_undefined) : value(type::k_undefined) {}
+value::value(const type id, stdx::string_view a, stdx::string_view b)
+    : _impl{stdx::make_unique<impl>()} {
+    switch (id) {
+        case type::k_regex:
+            _impl->_value.value_type = BSON_TYPE_REGEX;
+            _impl->_value.value.v_regex.regex = make_copy_for_libbson(a);
+            _impl->_value.value.v_regex.options = make_copy_for_libbson(b);
+            break;
+        case type::k_code:
+            _impl->_value.value_type = BSON_TYPE_CODE;
+            _impl->_value.value.v_code.code = make_copy_for_libbson(a);
+            _impl->_value.value.v_code.code_len = (uint32_t)a.length();
+            break;
+        case type::k_symbol:
+            _impl->_value.value_type = BSON_TYPE_SYMBOL;
+            _impl->_value.value.v_symbol.symbol = make_copy_for_libbson(a);
+            _impl->_value.value.v_symbol.len = (uint32_t)a.length();
+            break;
+        case type::k_minkey:
```

Comment:
Ah, I missed this response on my last pass. If `b` was required, this could be:
```
value(const type id, stdx::string_view a, stdx::string_view b); // regex with options
value(const type id, stdx::string_view a);                      // regex without options, symbol, code
```